### DISCUSSION TITLE
Mirgrate from carco to rsbuild, there are some errors.

### DISCUSSION THREAD

**Discussion by zouhangwithsweet** [web-infra-dev/rsbuild#487] [General]
run rsbuild start

```bash
error   Compile error: 
Failed to compile, check the errors for troubleshooting.
error[internal]: Failed to resolve /Users/xxxx/node_modules/@rspack/plugin-react-refresh/client/reactRefreshEntry.js in project root
```

open the localhost:8080
```html
Html Webpack Plugin:
Error: [html-rspack-plugin]: [html-rspack-plugin]: Failed to resolve /Users/xxxx/node_modules/@rsbuild/core/static/template.html in project root
```

Env:
macOS 13.3.1 (22E261)
MacBook M1
node v18.15.0

What could be the cause of this error?

**Comment by chenjiahan** [comment]
Could you provide a repro repository or more information to help us locate the problem? ❤️

**Comment by zouhangwithsweet** [comment]
Yeah. I want to help Uniswap to migrate to rsbuild, at https://github.com/Uniswap/interface. It's a big carco project. I failed at the beginning. :)

**Reply by zouhangwithsweet** [reply]
I do some thing like this

- add source entrys
- remove src/index.tsx to a pure comp with `react.createRoot`

All doesnot work.

**Reply by chenjiahan** [reply]
Get~ I will try to reproduce the error

**Reply by chenjiahan** [reply]
I tried to run `craco start` and get the following errors:
![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/e1abf2e2-739c-484f-944c-b16bbb903788)

It seems that we need to pull the graphql schema first, so I ran `yarn graphql:fetch` and the network paused:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/7573a32e-373b-4624-ac44-f100b08a8581)

Can you provide some steps to run the application?

**Reply by zouhangwithsweet** [reply]
src/graphql/thegraph/schema.graphql https://gist.github.com/zouhangwithsweet/83b74810a9f1e9ad997199078989dc63
src/graphql/data/schema.graphql https://gist.github.com/zouhangwithsweet/021b9f80371f594b853484d556f98a6a

**Reply by zouhangwithsweet** [reply]
copy the two graphql file to local project. Then run `yarn graphql:generate`

**Reply by chenjiahan** [reply]
get, I will try it tomorrow 👌

**Reply by chenjiahan** [reply]
I have reproduced the problem, and I will try to locate it next

**Reply by chenjiahan** [reply]
This comma in `tsconfig.json` breaks the Rspack resolver 😂 I will create an Rspack issue for this.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/4f6e49c7-447f-4291-9a31-625b6b323117)

**Reply by chenjiahan** [reply]
Rspack v0.3.12 has fixed this issue, I will release a new Rsbuild version soon.

https://github.com/web-infra-dev/rspack/releases/tag/v0.3.12

**Reply by chenjiahan** [reply]
Released: https://github.com/web-infra-dev/rsbuild/releases/tag/v0.0.22

**Reply by zouhangwithsweet** [reply]
Greate help!
